library/std/src/keyword_docs.rs

```diff
@@ -2388,6 +2410,40 @@ mod while_keyword {}
 ///
 /// We have written an [async book] detailing `async`/`await` and trade-offs compared to using threads.
 ///
+/// ## Control Flow
+/// [`return`] statements and [`?`][try operator] operators within `async` blocks do not cause
+/// a return from the parent function; rather, they cause the `Future` returned by the block to
+/// return with that value.
+///
+/// For example, the following Rust function will return `5`, assigning the `!` value to `x`, which
```

Comment:
*Assigning* is a little misleading here because `!` can never exist. Maybe something like "causing `x` to take the type of `!`".